classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -223,10 +229,80 @@ protected function checkUsageStatsLogs(): void
         ) {
             throw new Exception("There are one or more log files that were unable to finish processing. This happens when the scheduled task to process usage stats logs encounters a failure of some kind. These logs must be repaired and reprocessed or removed before the upgrade can continue. The logs can be found in the folders reject, processing and stage in {$usageStatsDir}.");
         }
+
+        $this->checkUsageStatsLogFileUrl($filePathToCheck);
+    }
+
+    /**
+     * Check if the domains in the log file URLs are correct i.e. the same as the base_url in the config.inc.php.
+     * Only the first URL in the log file is checked (and the same assumed for all URLs in the log file).
+     *
+     * @throws Exception
+     */
+    protected function checkUsageStatsLogFileUrl(string $filePathToCheck): void
+    {
+        $fhandle = fopen($filePathToCheck, 'r');
+        if (!$fhandle) {
+            throw new Exception("Can not open file {$filePathToCheck}.");
+        }
+        while (!feof($fhandle)) {
+            $line = trim(fgets($fhandle));
+            if (empty($line) || substr($line, 0, 1) === '#') {
+                continue;
+            } // Spacing or comment lines.
+
+            $entryData = $this->getDataFromLogEntry($line);
+
+            if (empty($entryData)) {
+                continue;
+            }
+            // Avoid internal apache requests.
+            if ($entryData['url'] == '*') {
+                continue;
+            }
+            // Avoid non sucessful requests.
+            $sucessfulReturnCodes = [200, 304];
+            if (!in_array($entryData['returnCode'], $sucessfulReturnCodes)) {
+                continue;
+            }
+
+            $configBaseUrl = Config::getVar('general', 'base_url');
+            $baseUrls = array_merge(Config::getContextBaseUrls(), [$configBaseUrl]);
+            foreach ($baseUrls as $baseUrl) {
+                if (str_contains($entryData['url'], $baseUrl)) {
+                    return;
+                }
+            }
+            break;
+        }
+        fclose($fhandle);
+        throw new Exception('The base_url in config.inc.php should be the same as in URLs in the usage stats log file.');
+    }
+
+    /**
+     * Get data from the passed log entry.
+     */
+    protected function getDataFromLogEntry(string $entry): array
+    {
+        $entryData = [];
+        // The default regex that can parse the usageStats plugin's log files.
+        $parseRegex = '/^(?P<ip>\S+) \S+ \S+ "(?P<date>.*?)" (?P<url>\S+) (?P<returnCode>\S+) "(?P<userAgent>.*?)"/';
+        if (preg_match($parseRegex, $entry, $m)) {
+            $associative = count(array_filter(array_keys($m), 'is_string')) > 0;
```

Comment:
This can be removed. Our version of PHP will always provide the named groups. And if it didn't, it would be better to use the integer indexed anyway.